plugins/nordic-hid/fu-nordic-hid-archive.c

```diff
@@ -11,15 +11,161 @@
 #include "fu-nordic-hid-firmware-b0.h"
 #include "fu-nordic-hid-firmware-mcuboot.h"
 
-/* current version format is 0 */
-#define MAX_VERSION_FORMAT 0
+/* the plugin currently supports version format of either 0 or 1 */
+#define MIN_VERSION_FORMAT 0
+#define MAX_VERSION_FORMAT 1
 
 struct _FuNordicHidArchive {
 	FuFirmwareClass parent_instance;
 };
 
 G_DEFINE_TYPE(FuNordicHidArchive, fu_nordic_hid_archive, FU_TYPE_FIRMWARE)
 
+static const gchar *
+fu_nordic_hid_archive_parse_file_get_bootloader_name(JsonObject *obj, GError **error)
+{
+	if (json_object_has_member(obj, "version_B0")) {
+		return "B0";
+	} else if (json_object_has_member(obj, "version_MCUBOOT")) {
+		return "MCUBOOT";
+	} else if (json_object_has_member(obj, "version_MCUBOOT+XIP")) {
+		return "MCUBOOT+XIP";
```

Comment:
if you do this Coverity gets upset with me. Just use

    if (watever)
        return foo;
    if (something)
        return bar;

thanks.

Reply:
Pushed changes. Is the updated approach with a temporary variable ok? If not then I will update according to this commit (TBH I prefer using temporary variable here)

Reply:
It's a bit weird tbh. Normally after `g_set_error` you see a return `FALSE`. Can you do the if (watever)
    return foo;` way please.

Reply:
I will switch to early returns and separate some more subfunctions. Then I will ask you for rereview.

The change will be delayed because I start a 2-week vacation tomorrow (I will ping you for rereview once PR will be ready). I want to test changes properly to ensure I would not break something accidentally.

Reply:
Excellent, thanks. Enjoy your holiday!